api/end_device.proto

```diff
@@ -69,9 +69,11 @@ message MACParameters {
   // ADR: number of retransmissions.
   uint32 adr_nb_trans = 6 [(gogoproto.customname) = "ADRNbTrans", (validate.rules).uint32.lte = 15];
   // ADR: number of messages to wait before setting ADRAckReq.
-  uint32 adr_ack_limit = 7 [(gogoproto.customname) = "ADRAckLimit", (validate.rules).uint32 = {gte: 1, lte: 32768}];
+  // This field is deprecated, use adr_ack_limit_exponent instead.
+  uint32 adr_ack_limit = 7 [(gogoproto.customname) = "DEPRECATED_ADRAckLimit"];
```

Comment:
I don't like that we're now generating `DEPRECATED_` fields in our Go code. The comment and `(deprecated) = true` option should be sufficient to indicate that a field is deprecated.

Reply:
What other suffix/prefix do you suggest?

Reply:
My suggestion is to leave the name of this field as `ADRAckLimit`, add the `(deprecated) = true` option and name the new field `ADRAckLimitExponent`.

Reply:
please see https://github.com/TheThingsNetwork/lorawan-stack/pull/1329#discussion_r326994634 why that's not a good solution.
What's your suggestion of naming deprecated fields when we have a name clash?

Reply:
I think we shouldn't rename old fields at all. That's just something that has to wait until v4.

Reply:
We can `reserve` deprecated fields if they're not used anymore at all. In this case, it seems to be used, and we cannot change rename.